### ISSUE TITLE
Enhance p:document-property() function?

### ISSUE THREAD

**Issue by eriksiegel** [xproc/3.0-specification#585]
Since we've changed the document properties into a `map(xs:QName, item()*)` it will be a nuisance to get anything out of this map because you always have to use the QName function to create a key and get something out. So I think it's a good idea to enhance the p:document-property() function:

Now: `p:document-property($doc as document-node(), $key as xs:QName) as item()` (the return type is wrong anyway I think)

Proposed: `p:document-property($doc as document-node(), $key as item()) as item()*`

Rules for interpreting $key are roughly the same as for p:force-qname-keys():

- If $key is of type xs:QName it is used unchanged
- If $key is of type xs:string it is transformed into an xs:Qname using the XPath EQName production rules. That is, it can be written as a local-name only, as a prefix plus local-name or as a URI plus local-name (using the Q{} syntax). It is a dynamic error (err:XD0061) if a string $key cannot be converted into a QName.
- If $key is of any other type, the function returns the empty sequence.

**Comment by xml-project** [comment]
Yes, definitely!!! I run across this last week and forgot to file an issue.
